bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/internal/SecureHttpClientFactory.java

```diff
@@ -0,0 +1,171 @@
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Factory class to create jetty http clients
+ *
+ * @author Michael Bock - initial API
+ */
+@Component(service = HttpClientFactory.class, immediate = true, name = "org.eclipse.smarthome.io.net.http.internal.SecureHttpClientFactory", configurationPid = "org.eclipse.smarthome.io.net.http.HttpClientFactory")
+public class SecureHttpClientFactory implements HttpClientFactory {
+
+    private final Logger logger = LoggerFactory.getLogger(SecureHttpClientFactory.class);
+
+    private static final String CONFIG_MIN_THREADS = "minThreads";
+    private static final String CONFIG_MAX_THREADS = "maxThreads";
+    private static final String CONFIG_KEEP_ALIVE = "keepAliveTimeout";
+    private static final String CONFIG_QUEUE_SIZE = "queueSize";
+
+    private static final int MIN_CONSUMER_NAME_LENGTH = 4;
+    private static final int MAX_CONSUMER_NAME_LENGTH = 20;
+    private static final Pattern CONSUMER_NAME_PATTERN = Pattern.compile("[a-zA-Z0-9_\\-]*");
+
+    private CustomTrustManagerProvider customTrustmanagerProvider;
+
+    private int minThreads;
+    private int maxThreads;
+    private int keepAliveTimeout;
+    private int queueSize;
+
+    private ThreadPoolExecutor threadPool;
+
+    @Activate
+    protected void activate(Map<String, Object> parameters) {
+        minThreads = (Integer) parameters.get(CONFIG_MIN_THREADS);
+        maxThreads = (Integer) parameters.get(CONFIG_MAX_THREADS);
+        keepAliveTimeout = (Integer) parameters.get(CONFIG_KEEP_ALIVE);
+        queueSize = (Integer) parameters.get(CONFIG_QUEUE_SIZE);
+        threadPool = new ThreadPoolExecutor(minThreads, maxThreads, keepAliveTimeout, TimeUnit.SECONDS,
+                new ArrayBlockingQueue<Runnable>(queueSize), new NamedThreadFactory("jetty"));
+        logger.info("jetty thread pool started with min threads {}, max threads {}", minThreads, maxThreads);
+    }
+
+    @Modified
+    protected void modified(Map<String, Object> parameters) {
+        minThreads = (Integer) parameters.get(CONFIG_MIN_THREADS);
+        maxThreads = (Integer) parameters.get(CONFIG_MAX_THREADS);
+        keepAliveTimeout = (Integer) parameters.get(CONFIG_KEEP_ALIVE);
+        queueSize = (Integer) parameters.get(CONFIG_QUEUE_SIZE);
+        threadPool.setCorePoolSize(minThreads);
+        threadPool.setMaximumPoolSize(maxThreads);
+        threadPool.setKeepAliveTime(keepAliveTimeout, TimeUnit.SECONDS);
+        logger.debug("jetty thread pool changed to min threads {}, max threads {}", minThreads, maxThreads);
+    }
+
+    @Deactivate
+    protected void deactivate() {
+        threadPool.shutdown();
```

Comment:
I believe we should use `ThreadPoolExecutor#shutdownNow()` to stop all actively executing tasks and the waiting tasks as well.